stack/eslint-plugin-packlets/src/Path.ts

```diff
@@ -2,9 +2,113 @@
 // See LICENSE in the project root for license information.
 
 import * as path from 'path';
+import * as fs from 'fs';
+
+export type ParsedPath = path.ParsedPath;
 
-// These helpers are borrowed from @rushstack/node-core-library
 export class Path {
+  /**
+   * Whether the filesystem is assumed to be case sensitive for Path operations.
+   *
+   * @remarks
+   * Regardless of operating system, a given file system's paths may be case-sensitive or case-insensitive.
+   * If a volume is mounted under a subfolder, then different parts of a path can even have different
+   * case-sensitivity.  The Node.js "path" API naively assumes that all Windows paths are case-insensitive,
+   * and that all other OS's are case-sensitive.  This is way off, for example a modern MacBook has a
+   * case-insensitive filesystem by default.  There isn't an easy workaround because Node.js does not expose
```

Comment:
@iclanton I'm amazed we didn't encounter this problem earlier.  It's very surprising.

Maybe we should consider deprecating `path` and apply this approach to `@rushstack/node-core-library`'s `Path` API.

Reply:
I'm a little surprised as well. I'm cool with deprecating Node's `path` as long as our thing's perf isn't worse.